### ISSUE TITLE
Cannot parse selector

### ISSUE THREAD

**Issue by Oluwasetemi** [stylelint/stylelint#5561]
> Clearly describe the bug
![image](https://user-images.githubusercontent.com/10030028/134694145-44e3ce64-18d2-4efd-9d96-ba7844be155c.png)

![image](https://user-images.githubusercontent.com/10030028/134693794-95df3802-f9ab-4d9e-867e-b284c251f94c.png)
The bug is caused on line 64: 3 considering this output which I assume is cased by `&:hover {  --cast: 6px }` line. The parser error points to the closing curly braces of `&:hover` part of the button. I cannot understand why this should be an error it is a valid css.

> Which rule, if any, is the bug related to?
It is a parseError.

> What code is needed to reproduce the bug?

```css
button {
    color: var(--white);
    border: 1px solid var(--color);
    background: var(--color-400);
    ${'' /* border: 0; */}
    padding: 0.6rem 1rem;
    border-radius: 2px;
    cursor: pointer;
    --cast: 2px;
    box-shadow: var(--cast) var(--cast) 0 var(--grey);
    text-shadow: 0.5px 0.5px 0 rgba(0,0,0,0.2);
    transition: all 0.2s;

    &:hover {
      --cast: 6px;
    }
  }
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "processors": [
    "stylelint-processor-styled-components"
  ],
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ]
}
```

> Which version of stylelint are you using?
```json
"stylelint": "^13.13.1",
  "stylelint-config-recommended": "^5.0.0",
  "stylelint-config-styled-components": "^0.1.1",
  "stylelint-processor-styled-components": "^1.10.0"
```

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

 "CLI with `stylelint './src/**/*.(ts|tsx|js|jsx)'`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

 No

> What did you expect to happen?

 "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
src/styles/GlobalStyles.js
 64:3  ✖  Cannot parse selector   parseError
 64:3  ✖  Cannot parse selector   parseError
 64:3  ✖  Cannot parse selector   parseError
```
I am using `"stylelint-processor-styled-components"` and the css is written in styled components.

**Comment by alexander-akait** [comment]
Bug in `stylelint-processor-styled-components`

**Comment by Oluwasetemi** [comment]
Thanks!!
